markers/sauce.py

```diff
@@ -18,3 +19,11 @@ def pytest_configure(config):
             config.option.markexpr = 'sauce and ({})'.format(config.option.markexpr)
         else:
             config.option.markexpr = 'sauce'
+
+
+def pytest_runtest_setup(item):
+    browser.ensure_browser_open(item)
+
+
+def pytest_runtest_teardown():
```

Comment:
the idea of quitting browser per test was to avoid timeout's on sauce dashboard. as there will be new browser launched per test. 
the right browser is opened or not is checked at ensure_browser_open()